src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditClientContactDetailsController.java

```diff
@@ -11,73 +10,80 @@
+import uk.gov.laa.ccms.caab.builders.DropdownBuilder;
 import uk.gov.laa.ccms.caab.service.CommonLookupService;
-import uk.gov.laa.ccms.data.model.CommonLookupDetail;
 
 /**
  * Controller for handling edits to client basic details during the application summary process.
  */
 @Controller
 @Slf4j
 @RequiredArgsConstructor
-@SessionAttributes({CLIENT_DETAILS})
+@SessionAttributes({CLIENT_FLOW_FORM_DATA})
 public class EditClientContactDetailsController {
 
   private final CommonLookupService commonLookupService;
 
   private final ClientContactDetailsValidator clientContactDetailsValidator;
 
-  @ModelAttribute("editedClientDetails")
-  public ClientDetails getClientDetails(HttpSession session) {
-    return (ClientDetails) session.getAttribute(CLIENT_DETAILS);
+  @ModelAttribute("contactDetails")
+  public ClientFormDataContactDetails getContactDetails() {
+    return new ClientFormDataContactDetails();
   }
 
   /**
    * Handles the GET request for edit client contact details page.
    *
-   * @param clientDetails The details of the client.
+   * @param clientFlowFormData The data for client flow.
    * @param model The model for the view.
    * @return The view name for the client basic details page
    */
   @GetMapping("/application/summary/client/details/contact")
   public String getClientDetailsBasic(
-      @ModelAttribute("editedClientDetails") ClientDetails clientDetails,
+      @SessionAttribute(CLIENT_FLOW_FORM_DATA) ClientFlowFormData clientFlowFormData,
       Model model) {
 
     populateDropdowns(model);
-    model.addAttribute("editedClientDetails", clientDetails);
+    ClientFormDataContactDetails contactDetails = clientFlowFormData.getContactDetails();
+    contactDetails.setClientFlowFormAction(clientFlowFormData.getAction());
+
+    model.addAttribute("contactDetails", contactDetails);
 
     return "application/summary/client-contact-details";
   }
 
   /**
    * Handles the edit contact client details submission.
    *
-   * @param clientDetails The details of the client.
+   * @param clientFlowFormData The data for create client flow.
+   * @param contactDetails The contact details of the client.
    * @param bindingResult Validation result.
    * @param model The model for the view.
    * @return A redirect string to the agreement page.
    */
   @PostMapping("/application/summary/client/details/contact")
   public String postClientDetailsBasic(
-      @ModelAttribute("editedClientDetails") ClientDetails clientDetails,
+      @SessionAttribute(CLIENT_FLOW_FORM_DATA) ClientFlowFormData clientFlowFormData,
+      @ModelAttribute("contactDetails") ClientFormDataContactDetails contactDetails,
       BindingResult bindingResult,
-      Model model,
-      HttpSession session) {
+      Model model) {
 
-    clientContactDetailsValidator.validate(clientDetails, bindingResult);
+    contactDetails.setPassword(clientFlowFormData.getContactDetails().getPassword());
```

Comment:
Not sure I understand this. Is this not overwriting whatever the user just posted for password?

Reply:
On the edit page - password is hidden away and cannot be changed in the PUI.
I didnt think it was suitable to add a hidden field for the user to override it.
added it here so that the validator passes and that the password retains its original value.